Adding Starbucks portal and portal scraper

RocketGod-git opened the pull request:
Fixed. Thanks @amouly for the tip.

RocketGod-git committed:
Adding Starbucks portal and portal scraper

This is not ready to merge, but I need help making the form typable. It's giving me a headache. :angry:

Their website's sign in portal is massive amounts of js only a maniac could go through. So I made a portal scraper also in this PR which rips their website, saves images local, and replaces those links with the new one. It also removes things like google analytics automatically.

Only problem I have is getting the damn form to allow input in the input boxes.

Feel free to fix or tell me to eff off. :wink:

amouly commented:
Why would you do all this work? you can simply install Chrome extension and download the file of that website:

https://chrome.google.com/webstore/detail/singlefile/mpiodijhokgodhhofbcjdecpffjipkle/related

amouly commented on portals/Starbucks/portalscraper/chromedriver.exe:
This only works for Windows, please avoid adding this dependency.

RocketGod-git replied:
No problem. I'm sure there's another way.

RocketGod-git replied:
Updated the PR to include Linux.

amouly replied:
@RocketGod-git still doesn't make sense to me.

I don't use Windows or Linux, why I would need to download those executables?

This should be an external dependency, installed from a dependency manager or something smart that allows me to install only the dependencies for my OS.

This project it's about an application to host an "evil portal", so you don't need to version also the tool to create the portals, that should be independent.

RocketGod-git replied:
I'll just remove it for now until finished...

RocketGod-git committed:
Fixed for Linux or Windows use

RocketGod-git committed:
Removed

RocketGod-git commented:
lol. Never tried sneaking company website logins before. So ya, I never saw this extension. I'll try that. Thanks.

RocketGod-git committed:
Fixed

Thanks, Adrian Mouly :smile:

RogueMaster commented:
I think there is a file limit on the portal size, not sure exactly what it is, but, this file is much larger than the other portals.

RocketGod-git commented:
Ya... I'll remove as much js as I can 🧐

RogueMaster commented:
Limit is 20kb

RocketGod-git commented:
I thought it was js carried over, but it's the images embedded with base64...

RocketGod-git committed:
Replaced all base64 and useless css and metadata

Got it down to 49kb. I don't see any way to squeeze it more.

RogueMaster commented:
Down to 48.2KB, still too big 😢 Maybe something like [tinyimg](https://tinypng.com/) can help...

RocketGod-git commented:
Seems impossible to halve this 😒 lol. I'll try some more in a bit.

RocketGod-git committed:
Update Starbucks.html

RocketGod-git committed:
Compressed further.

I got it down to 38kb. Any suggestions on compressing the svg data would be great.

RocketGod-git committed:
19KB

I managed to get it under 20kb. Please review.

RocketGod-git commented:
Got it to 19KB but couldn't keep the logo. Still looks good with the rest of the elements I could keep. Made a tool to help squash the html. 
[https://github.com/RocketGod-git/evilportal-htmlsquash](https://github.com/RocketGod-git/evilportal-htmlsquash)

RocketGod-git committed:
Create Caesars_Rewards.html

RocketGod-git committed:
Fixed form action.
